lib/BoundaryClickWatcher/README.md

```diff
@@ -0,0 +1,40 @@
+#BoundaryClickWatcher
+Wrap a component in BoundaryClickWatcher to detect clicks on our outside of the component. See an example of this in the Conversation story.
+
+##Usage
+
+BoundaryClickWatcher has a boolean (boundaryIsActive) which is set to true when it detects an inner click and back to false when it detects a click outside of the boundary. The value of this can be passed onto the child component like so:
+
+```
+<BoundaryClickWatcher>
+  {(boundaryIsActive) => {
+    return (
+      <My Component
+        boolProp={boundaryIsActive}
+      />
+    )
+  }}
+</BoundaryClickWatcher>
+```
+
+*(Optional)* It has two props `insideClickHandler` and `outsideClickHandler` you can pass in actions for added custom behaviour on inner or outer click.
```

Comment:
I wonder if we should make this more concise?

Something like the following section in a different library: https://github.com/igorprado/react-notification-system#methods

e.g.

--------

## Props

### `insideClickHandler`

Executes when the user clicks inside the boundary.

### ```outsideClickHandler```

Executes when the user clicks outside the boundary.

--------

Or maybe a table like https://github.com/igorprado/react-notification-system/blob/master/README.md#creating-a-notification

What do you think?